src/lib.rs

```diff
@@ -43,7 +43,7 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 #![cfg_attr(all(not(feature = "std"), feature = "error"), feature(error_in_core))]
 #![cfg_attr(docsrs, feature(doc_auto_cfg))]
-#![deny(rustdoc::broken_intra_doc_links, rustdoc::private_intra_doc_links)]
+#![deny(rustdoc::all)]
```

Comment:
I feel like we want to do this for dev and CI, but not when docs.rs is actually building the docs. Otherwise it might fail to do so if some new lint gets added. Let's instead add `-D rustdoc::all` to the `RUSTDOCFLAGS` in CI.

Reply:
@JelteF 

> otherwise it might fail to do so if some new lint gets added

How is it possible? Build on docs.rs is done right after CI passed and crate is released.

> Let's instead add `-D rustdoc::all` to the `RUSTDOCFLAGS` in CI. (we should double check that it works)

Don't really like the idea of additional flags here and there.

Maybe `#![cfg_attr(not(docsrs), deny(rustdoc::all))]` would do the trick?

Reply:
> How is it possible? Build on docs.rs is done right after CI passed and crate is released.

I'm pretty sure docs.rs might rebuild docs later again after a new release of rustdoc (e.g. to get new styling).

> Maybe `#![cfg_attr(not(docsrs), deny(rustdoc::all))]` would do the trick?

I guess that would work, but would want to re-enable the the errors for CI so probably something like:
```
#![cfg_attr(any(not(docsrs), ci), deny(rustdoc::all))]
```

And then adding `--cfg ci` to the RUSTDOCFLAGS in CI.

Reply:
@JelteF 

> I'm pretty sure docs.rs might rebuild docs later again after a new release of rustdoc (e.g. to get new styling).

Never heard of that. In my experience, once docs are built, they remain as they were built. Having all the old style forever. You can see the old style on old crates, like <https://docs.rs/iron/0.4.0>.

> but would want to re-enable the the errors for CI so probably something like

👍